text/entities/0264-resource-and-entities.md

```diff
@@ -0,0 +1,863 @@
+### How to attach Entity "bundle" information in Resource?
+
+The protocol today requires a raw grab bag of Attributes on Resource. We cannot break this going forward.  However, Entities represent a new mechanism of "bundling" attributes on Resource and interacting with these bundles.  We do not want this to bloat the protocol, nor do we want it to cause oddities.
+
+Going forward, we have set of options:
+
+- Duplicate attributes in `Entity` section of Resource.
+- Reference attributes of Resource in entity.
+- Only identify Entity id and keep attribute<->entity association out of band.
+- Extend Attribute on Resource so that we can track the entity type per Key-Value (across any attribute in OTLP).
+
+The third option prevents generic code from interacting with Resource and Entity without understanding the model of each.  The first keeps all usage of entity simple at the expense of duplicating information and the middle is awkward to interact with from an OTLP usage perspective. The fourth is violates our stability policy for OTLP.
+
+### How to deal with Resource/Entities whose lifecycle does not match the SDK?
+
+This proposal motivates a Resource Provider in the SDK whose job could include managing changes in entity lifetimes, but does not account for how these changes would be broadcast across TracerProvider, LogProvider, MeterProvider, etc.  That would be addressed in a follow on OTEP.
+
+### How to deal with Prometheus Compatibility for non-SDK telemetry?
+
+Today, [Prometheus compatibility](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/compatibility/prometheus_and_openmetrics.md) relies on two key attributes in Resource: `service.name` and `service.instance.id`. These are not guaranteed to exist outside of OpenTelemetry SDK generation. While this question is not fully answered, we believe outlining identity in all resources within OpenTelemetry allows us to define a solution in the future while preserving compatibility with what works today.
+
+Here's a list of requirements for the solution:
+
+- Existing prometheus/OpenTelemetry users should be able to migrate from where they are today.
+- Any solution MUST work with the [info-typed metrics](https://github.com/prometheus/proposals/blob/main/proposals/2024-04-10-native-support-for-info-metrics-metadata.md#goals) being added in prometheus.
+  - Resource descriptive attributes should leverage `info()` or metadata.
+  - Resource identifying attributes need more thought/design from OpenTelemetry semconv + entities WG.
+  - Note: Current `info()` design will only work with `target_info` metric, and `job/instance` labels for joins.  This labels MUST be generated by the OTLP endpoint in prometheus.
+- (desired) Users should be able to correlate metric timeseries to other signals via Resource attributes showing up as labels in prometheus.
+- (desired) Conversion from `OTLP -> prometheus` can be reversed such that `OTLP -> Prometheus -> OTLP` is non-lossy.
```

Comment:
> - (desired) Conversion from `OTLP -> prometheus` can be reversed such that `OTLP -> Prometheus -> OTLP` is non-lossy.

I'm bringing that up to the Prometheus team. We've recently decided that [Prometheus won't support exporting OTLP directly](https://github.com/prometheus/prometheus/issues/12633#issuecomment-2359054297), but we're working on [Prometheus Remote Write Receiver in OTel collector](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35295) and that's our plan to transform Prometheus metrics into OTLP format, would that be ok?

Reply:
Yes - to me it's the same domain modelling issue